fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexScrubbingToolsMissing.java

```diff
@@ -47,20 +48,20 @@
 import java.io.IOException;
 import java.util.Collection;
 import java.util.Collections;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.Optional;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicReference;
 import java.util.stream.Collectors;
 
 /**
  * Index Scrubbing Toolbox for a Lucene index maintainer. Scrub missing value index entries - i.e. detect record(s) that should
- * cannot be found in the segment index.
+ * have been indexed, but cannot be found in the segment index.
  */
-public class LuceneIndexScrubbingToolsMissing implements IndexScrubbingTools<FDBStoredRecord<Message>> {
+public class LuceneIndexScrubbingToolsMissing extends ValueIndexScrubbingToolsMissing {
```

Comment:
I think I agree with your comment that extending `ValueIndexScrubbingToolsMissing` feels wrong.
Perhaps a `BaseScrubbingToolsMissing` would make sense, although I think the only method that actually should be shared is `getCursor`, so probably a utility class would make more sense.

Leaving as you have it until we have a third `Missing` implementation also seems reasonable, as it might also align with better abstracting synthetic records in general for use across scrubbing, indexing and IndexMaintenance.